src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests4.cs

```diff
@@ -1487,5 +1487,282 @@ static int M1(bool? b1, bool? b2)
                 Diagnostic(ErrorCode.ERR_SwitchArmSubsumed, "(null, true)").WithLocation(13, 13)
                 );
         }
+
+        [Fact]
+        public void WrongNumberOfDesignatorsForTuple()
+        {
+            var source =
+@"class Program
+{
+    static void Main()
+    {
+        _ = (1, 2) is var (_, _, _);
+    }
+}
+";
+            var compilation = CreatePatternCompilation(source);
+            compilation.VerifyDiagnostics(
+                // (5,27): error CS8502: Matching the tuple type '(int, int)' requires '2' subpatterns, but '3' subpatterns are present.
+                //         _ = (1, 2) is var (_, _, _);
+                Diagnostic(ErrorCode.ERR_WrongNumberOfSubpatterns, "(_, _, _)").WithArguments("(int, int)", "2", "3").WithLocation(5, 27)
+                );
+        }
+
+        [Fact]
+        public void PropertyNameMissing()
+        {
+            var source =
+@"class Program
+{
+    static void Main()
+    {
+        _ = (1, 2) is { 1, 2 };
+    }
+}
+";
+            var compilation = CreatePatternCompilation(source);
+            compilation.VerifyDiagnostics(
+                // (5,25): error CS8503: A property subpattern requires a reference to the property or field to be matched, e.g. '{ Name: 1 }'
+                //         _ = (1, 2) is { 1, 2 };
+                Diagnostic(ErrorCode.ERR_PropertyPatternNameMissing, "1").WithArguments("1").WithLocation(5, 25)
+                );
+        }
+
+        [Fact]
+        public void IndexedProperty_01()
+        {
+            var source1 =
+@"Imports System
+Imports System.Runtime.InteropServices
+<Assembly: PrimaryInteropAssembly(0, 0)> 
+<Assembly: Guid(""165F752D-E9C4-4F7E-B0D0-CDFD7A36E210"")> 
+<ComImport()>
+<Guid(""165F752D-E9C4-4F7E-B0D0-CDFD7A36E211"")>
+Public Interface I
+    Property P(x As Object, Optional y As Object = Nothing) As Object
+End Interface";
+            var reference1 = BasicCompilationUtils.CompileToMetadata(source1);
+            var source2 =
+@"class C
+{
+    static void Main(I i)
+    {
+        _ = i is { P: 1 };
+    }
+}";
+            var compilation2 = CreateCompilation(source2, new[] { reference1 });
+            compilation2.VerifyDiagnostics(
+                // (5,20): error CS0857: Indexed property 'I.P' must have all arguments optional
+                //         _ = i is { P: 1 };
+                Diagnostic(ErrorCode.ERR_IndexedPropertyMustHaveAllOptionalParams, "P").WithArguments("I.P").WithLocation(5, 20)
+                );
+        }
+
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/31209")]
```

Comment:
I will unskip this test to record the current behavior (an error without crashing).